polygeist/tools/cgeist/Test/Verification/sycl/array-capture.cpp

```diff
@@ -57,7 +58,10 @@ int main(){
 // CHECK-NEXT:        %[[VAL_372:.*]] = "polygeist.subindex"(%[[VAL_358]], %[[VAL_361]]) : (memref<?x!llvm.struct<(!sycl_accessor_1_f32_rw_dev, array<8 x f32>)>, 4>, index) -> memref<?x!sycl_accessor_1_f32_rw_dev, 4>
 // CHECK-NEXT:        %[[VAL_373:.*]] = memref.memory_space_cast %[[VAL_365]] : memref<?x!sycl_id_1_> to memref<?x!sycl_id_1_, 4>
 // CHECK-NEXT:        %[[VAL_374:.*]] = memref.memory_space_cast %[[VAL_359]] : memref<?x!sycl_id_1_> to memref<?x!sycl_id_1_, 4>
-// CHECK-NEXT:        sycl.constructor @id(%[[VAL_373]], %[[VAL_374]]) {MangledFunctionName = @_ZN4sycl3_V12idILi1EEC1ERKS2_} : (memref<?x!sycl_id_1_, 4>, memref<?x!sycl_id_1_, 4>)
+// CHECK-NEXT:        %[[VAL_405:.*]] = sycl.id.constructor(%[[VAL_374]]) : (memref<?x!sycl_id_1_, 4>) -> memref<?x!sycl_id_1_, 4>
+// CHECK-NEXT:        %[[VAL_406:.*]] = "polygeist.memref2pointer"(%[[VAL_373]]) : (memref<?x!sycl_id_1_, 4>) -> !llvm.ptr<4>
+// CHECK-NEXT:        %[[VAL_407:.*]] = "polygeist.memref2pointer"(%[[VAL_405]]) : (memref<?x!sycl_id_1_, 4>) -> !llvm.ptr<4>
+// CHECK-NEXT:        "llvm.intr.memcpy"(%[[VAL_406]], %[[VAL_407]], %[[SIZE]]) <{isVolatile = false}> : (!llvm.ptr<4>, !llvm.ptr<4>, i64) -> ()
```

Comment:
This will be lowered and optimized to a single `memcpy` intrinsic. We need the overhead for now due to SYCL dialect lacking "memref semantics" constructors